server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTarget.java

```diff
@@ -213,23 +211,12 @@ private void finalizeReplication(CheckpointInfoResponse checkpointInfoResponse,
         try {
             store = store();
             store.incRef();
-            Map<String, String> tempFileNames;
-            if (this.indexShard.indexSettings().isRemoteStoreEnabled() == true) {
-                tempFileNames = getSegmentFilesResponse.getFiles()
-                    .stream()
-                    .collect(Collectors.toMap(StoreFileMetadata::name, StoreFileMetadata::name));
-            } else {
-                tempFileNames = multiFileWriter.getTempFileNames();
-            }
-            store.buildInfosFromBytes(
-                tempFileNames,
+            multiFileWriter.renameAllTempFiles();
```

Comment:
its not used but this class still wires up a mfw and will basically no-op because there are no temp files.  I'd like to refactor this further to remove the node-node specific components and avoid introducing more conditional logic.

Reply:
Thanks @mch2, do we have tracking issue for this refactor.

Reply:
I think we can reuse https://github.com/opensearch-project/OpenSearch/issues/9016 here.